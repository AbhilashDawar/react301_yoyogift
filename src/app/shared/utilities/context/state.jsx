import * as React from 'react';

let ContextState = React.createContext();

let initialState = {
    showLoader: false,
    // filterParams: {
    //     startValue: config.rangeDefaults.min,
    //     endValue: config.rangeDefaults.max,
    //     discountOption: 0
    // }
};

let reducer = (state, action) => {
    switch (action.type) {
        case "RESET":
            return initialState;
        case "SHOW":
            return { ...state, showLoader: true };
        case "HIDE":
            return { ...state, showLoader: false };
        // case "UPDATE_FILTERS":
        //     return { ...state, filterParams: action.payload };
        default:
            return { ...state };
    }
};

function ContextStateProvider(props) {
    let [state, dispatch] = React.useReducer(reducer, initialState);
    let value = { state, dispatch };

    return (
        <ContextState.Provider value={value}>{props.children}</ContextState.Provider>
    );
}

let ContextStateConsumer = ContextState.Consumer;

export { ContextState, ContextStateProvider, ContextStateConsumer };