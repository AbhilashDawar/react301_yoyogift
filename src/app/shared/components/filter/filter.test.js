import React from 'react';
import AppFilter from './index.jsx';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ContextStateProvider } from '../../utilities/context/state.jsx';

test('AppFilter Component', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <ContextStateProvider>
                <AppFilter />
            </ContextStateProvider>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});