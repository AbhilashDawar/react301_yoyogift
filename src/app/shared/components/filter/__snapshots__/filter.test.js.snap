// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFilter Component 1`] = `
<div
  className="MuiGrid-root makeStyles-filterContainer-4 MuiGrid-container"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root makeStyles-expansionPanel-5 MuiExpansionPanel-rounded MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-expansionPanelSummary-6"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
        >
          Filter
        </h6>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
              >
                <div
                  className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-1 MuiPaper-rounded"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
